capstone-frontend/src/components/ClubPage.jsx

```diff
@@ -57,16 +70,23 @@ class ClubPage extends Component {
           });
     }
   }
- 
-  handleClick = () => {
-    const history = this.props.history;
-    fetch(`/api/clubs?id=${this.props.match.params.id}`, {method: "delete"})
-        .then(function() {
-            history.push("/myclubs");
-        })
+
+  handleAssignmentPost = (e) => {
+    e.preventDefault();
+    if (window.localStorage.getItem("userID") !== this.state.club.ownerID) {
+      alert("Assignment creation failed. You do not own this club.");
+      return;
+    }
+    let data = {
+      "clubID": this.state.club.id,
+      "text": e.target[0].value,
+      "whenCreated": (new Date()).toUTCString()
+    };
+    fetch(`/api/assignments`, {method: "post", body: JSON.stringify(data)})
+        .then(this.fetchData())
```

Comment:
Is it necessary to call `fetchData` when an assignment is posted? Wondering if you could get away with just making a call to the assignments servlet here to update the club page after a new assignment is posted.

Reply:
What I could do is break `fetchData` into several helper methods (ie. one for the club, one for assignments, one for the book, etc) this way I can just call `fetchAssignmentData` after an assignment post and re-fetch just the assignments rather than all the data for the whole page. 

I think an even better solution is to take the response json (which will be the created assignment) and append it to the state's assignments without re-fetching to trigger a re-render. This will be much quicker than refetching all the club data on a post, and was the reason I originally had the post method return the created object. I'll go with this option.

Reply:
I'm going to implement this same logic in my `onComment` function, as I was previously re-fetching all an assignment's comments onComment. Now I'll use the response from the post to update the state appropriately and reduce unnecessary fetches.